activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -1946,6 +1949,15 @@ def test_restrict_with_error
     assert firm.companies.exists?(name: "child")
   end
 
+  def test_restrict_with_error_through_accepts_nested_attributes_for
+    question = Question.create!
+    choice = Choice.create!(question: question)
+    Answer.create!(choice: choice)
+    question.assign_attributes(choices_attributes: [{ id: choice.id, _destroy: true }])
+
+    assert_nothing_raised { question.save }
```

Comment:
It would be good to assert that there are `errors` on the `question` object, similar to the test above.

Reply:
I added more expectations on this test (presence of records, presence of errors, expected error message) and an additional test to make sure this change does not affect how `restrict_with_exception` behaves.